src/components/RoverConnection.js

```diff
@@ -0,0 +1,105 @@
+import React, { Component, Fragment } from 'react';
+import { Card, Label, Icon } from 'semantic-ui-react';
+import { hot } from 'react-hot-loader';
+import PropTypes from 'prop-types';
+import Websocket from 'react-websocket';
+import { COVERED, NOT_COVERED } from '@/actions/sensor';
+
+const heartbeatTimeout = 8000; // milliseconds
+
+class RoverConnection extends Component {
+  constructor(props) {
+    super(props);
+
+    this.timeout = null;
+    this.state = {
+      online: false,
+    };
+  }
+
+  componentDidMount() {
+    this.startHeartbeatTimer();
+  }
+
+  startHeartbeatTimer = () => {
+    this.timeout = setTimeout(this.setOffline, heartbeatTimeout);
+  }
+
+  setActive = () => {
+    const { changeActiveRover, clientId } = this.props;
+
+    changeActiveRover(clientId);
+  }
+
+  setOffline = () => this.setState({ online: false })
+
+  setOnline = () => this.setState({ online: true })
+
+  onMessage = (data) => {
+    const { changeLeftSensorState, changeRightSensorState } = this.props;
+    const { online } = this.state;
+
+    const message = JSON.parse(data);
+
+    if (message.type === 'sensor-reading') {
```

Comment:
This also needs to check `isActive` before making any sensor changes. It would be good to add a test to verify that check as well